bundles/org.openhab.binding.hive/src/main/java/org/openhab/binding/hive/internal/client/adapter/ZonedDateTimeGsonAdapter.java

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.hive.internal.client.adapter;
+
+import java.io.IOException;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+
+import com.google.gson.stream.JsonReader;
+import com.google.gson.stream.JsonWriter;
+
+/**
+ * A gson {@link com.google.gson.TypeAdapter} for {@link ZonedDateTime} as used by the
+ * Hive API.
+ *
+ * @author Ross Brown - Initial contribution
+ */
+@NonNullByDefault
+public final class ZonedDateTimeGsonAdapter extends GsonTypeAdapterBase<ZonedDateTime> {
+    private static final DateTimeFormatter HIVE_DATETIME_FORMAT = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSSZ");
+
+    @Override
+    public void writeValue(final JsonWriter out, final @Nullable ZonedDateTime zonedDateTime) throws IOException {
+        if (zonedDateTime != null) {
+            out.value(zonedDateTime.format(HIVE_DATETIME_FORMAT));
+        } else {
+            out.nullValue();
+        }
+    }
+
+    @Override
+    public @Nullable ZonedDateTime readValue(final JsonReader in) throws IOException {
+        return ZonedDateTime.parse(in.nextString(), HIVE_DATETIME_FORMAT);
```

Comment:
Isn't this inconsistent with your `writeValue` function?

Reply:
Do you mean in the handling of null?

I don't expect to ever read null values so I didn't bother handling it and let `in.nextString()` throw its `IllegalStateException`

`writeValue` isn't ever used for ZonedDateTime. It could just throw `UnsupportedOperationException` and be fine. I guess I just provided an implementation because it was so simple.